victim-loss-identifier/src/fetcher.ts

```diff
@@ -0,0 +1,263 @@
+import { providers } from "ethers";
+import { apiKeys, Erc721Transfer, coinData } from "./types";
+import {
+  Network,
+  Alchemy,
+  GetFloorPriceResponse,
+  FloorPriceMarketplace,
+} from "alchemy-sdk";
+import { LRUCache } from "lru-cache";
+
+export default class DataFetcher {
+  provider: providers.Provider;
+  private apiKeys: apiKeys;
+  private alchemy: Alchemy;
+  private ethPriceCache: LRUCache<number, number>;
+  private floorPriceCache: LRUCache<number, number>;
+
+  constructor(provider: providers.Provider, apiKeys: apiKeys) {
+    this.apiKeys = apiKeys;
+    this.provider = provider;
+    this.alchemy = new Alchemy({
+      apiKey: this.apiKeys.victimLossKeys.alchemyApiKey,
+      network: Network.ETH_MAINNET,
+    });
+    this.ethPriceCache = new LRUCache<number, number>({
+      max: 1000,
+    });
+    this.floorPriceCache = new LRUCache<number, number>({
+      max: 10000,
+    });
+  }
+
+  getTransactionReceipt = async (txHash: string) => {
+    let receipt;
+    let tries = 0;
+    const maxTries = 3;
+
+    while (tries < maxTries) {
+      try {
+        receipt = await this.provider.getTransactionReceipt(txHash);
+        break; // exit the loop if successful
+      } catch (err) {
+        tries++;
+        if (tries === maxTries) {
+          throw err; // throw the error if maximum tries reached
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000)); // wait for 1 second before retrying
+      }
+    }
+    return receipt;
+  };
+
+  getTransaction = async (txHash: string) => {
+    let receipt;
+    let tries = 0;
+    const maxTries = 3;
```

Comment:
Small suggestion that `maxTries` could be moved one scope above, so not to repeat in each getter function. Could also be named to `MAX_TRIES` since it is a constant.